user/pages/03.commerce2/02.developer-guide/03.core/01.conditions/docs.md

```diff
@@ -4,8 +4,284 @@ taxonomy:
+`'#value'` in the `buildConfigurationForm` function. Doing so will stop any new
+configuration values set by user input from being saved. Use 
+`defaultConfiguration` function to set default values. 
+
+
+Default Configuration Settings
+---
+
+Emphasis of setting name `***'operator'***` is to show how this name should be 
+used across all functions in this class. 
+
+```
+public function defaultConfiguration() {
+  return [
+    ***'operator'*** => '>',
+    'quantity' => 1,
+  ] + parent::defaultConfiguration();
+}
+```
+
+
+Submit Configuration Form
+---
+
+This is where user inputted values are saved to the form configuration settings.
+
+```
+public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+  parent::submitConfigurationForm($form, $form_state);
+
+  $values = $form_state->getValue($form['#parents']);
+  
+// Note that the setting name here needs to be consistent with defaultConfig
+// and buildConfigurationForm.
+  $this->configuration[***'operator'***] = $values[***'operator'***];
+  $this->configuration['quantity'] = $values['quantity'];
+}
+```
+
+
+Evaluate Function
+---
+
+This is the guts of a condition. The goal of `evaluate` is to decide if an 
+order meets a certain criteria and to return a boolean TRUE if so. 
+
+```
+// *** Remember annotations are inherited from ConditionInterface.***
+public function evaluate(EntityInterface $entity) {
+
+// This line is essential as it ensures this condition will only run for the 
+// 'entity_type' declared in this condition's annotations. 
+  $this->assertEntity($entity);
+  
+  /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
+  $order = $entity;
+  /** @var \Drupal\commerce_promotion\Entity\PromotionInterface $promotion */
+  $promotion = $this->parentEntity;
+  $offer = $promotion->getOffer();
+
+  $quantity = '0';
+  foreach ($order->getItems() as $order_item) {
+  
+// First assess if each order item can apply to this condition.
+    // If the offer has conditions, skip order items that don't match.
+    if ($offer instanceof OrderItemPromotionOfferInterface) {
+    
+      $conditions = $offer->getConditions();
+      $condition_group = new ConditionGroup($conditions, 'OR');
+      
+      if (!$condition_group->evaluate($order_item)) {
+        continue;
+      }
+    }
+    
+// Get the quantity from each item and add to a running total.
+    $quantity = Calculator::add($quantity, $order_item->getQuantity());
+  }
+
+// This is where the final decision is made to whether a condition is met.
+  switch ($this->configuration['operator']) {
+    case '>=':
+     return $quantity >= $this->configuration['quantity'];
+     
+/* To be more explicit for this tutorial's use case: 
+ * If the total quantity of all order items is greater than or equal to the 
+ * quantity that the user set in the configuration form, return TRUE. If not, 
+ * return FALSE and the promotion will not be applied to the order.
+ */
```

Comment:
As above I'd leave out most of all of these explanations for the obvious stuff - the Calculator:add bit is self-explanatory and this following bit it just knowing what the comparison operators do.

Reply:
Also, this comment style is not Drupal code style friendly.